mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -0,0 +1,445 @@
+                                 outputDirectory, year)
+            if not os.path.exists(outputFileTseries):
+                for nm in range(12):
+                    print '     month = ', nm + 1
+                    horizontalVel = ds.avgNormalVelocity[ncount, :, :].values
+                    verticalVel = ds.avgVertVelocityTop[ncount, :, :].values
+                    velArea = verticalVel * areaCell[:, np.newaxis]
+
+                    transportZEdge = np.zeros([nVertLevels, maxEdgesInTransect])
+                    for i in range(maxEdgesInTransect):
+                        if transectEdgeGlobalIDs[i] == 0:
+                            break
+                        iEdge = transectEdgeGlobalIDs[i] - 1
+                        transportZEdge[:, i] = horizontalVel[iEdge, :] * \
+                            transectEdgeMaskSigns[iEdge, np.newaxis] * \
+                            dvEdge[iEdge, np.newaxis] * \
+                            refLayerThickness[np.newaxis, :]
+                        transportZ = transportZEdge.sum(axis=1)
+
+                    mocTop = np.zeros([np.size(mocLatRegional), nVertLevels+1])
+                    mocTop[0, range(1, nVertLevels+1)] = transportZ.cumsum()
+                    for iLat in range(1, np.size(mocLatRegional)):
+                        indlat = np.logical_and(np.logical_and(
+                                     regionCellMask == 1,
+                                     latCell >= mocLatRegional[iLat-1]),
+                                     latCell < mocLatRegional[iLat])
+                        mocTop[iLat, :] = mocTop[iLat-1, :] + \
+                                          velArea[indlat, :].sum(axis=0)
+                    # convert m^3/s to Sverdrup
+                    mocRegional = mocTop * m3ps_to_Sv
+                    mocRegional26[ncount] = np.amax(mocRegional[indLat26, :])
+                    ncount = ncount + 1
+
+                # Save to file
+                print '   save Atlantic MOC at 26.5N to yearly file...'
+                ncFile = netCDF4.Dataset(outputFileTseries, mode='w')
+                # create dimension
+                ncFile.createDimension('Time', 0)
+                # create variables
+                time = ncFile.createVariable('time', 'f4', ('Time',))
+                time.description = 'days since start of simulation'
+                fldAtlantic26 = ncFile.createVariable('mocRapidArrayMax',
+                                                      'f4', ('Time',))
+                fldAtlantic26.description = 'Max MOC Atlantic streamfunction'\
+                                            ' nearest to RAPID Array latitude'\
+                                            ' (26.5N)'
+                fldAtlantic26.units = 'Sv (10^6 m^3/s)'
+                # save variables
+                time[:] = ds.Time[ncount-12:ncount].values
+                fldAtlantic26[:] = mocRegional26[ncount-12:ncount]
+                ncFile.close()
+            else:
```

Comment:
Large if/else loops are difficult to understand.  This can be clarified by using more functions as possible to keep code in if/else sections to some small number of lines, e.g., <<50.